pkg/scraper/client/resource/decode.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2020 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"time"
+
+	"github.com/prometheus/common/model"
+	apitypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+
+	"sigs.k8s.io/metrics-server/pkg/storage"
+)
+
+const (
+	nameSpaceMetricName         = "namespace"
+	podNameMetricName           = "pod"
+	containerNameMetricName     = "container"
+	nodeCpuUsageMetricName      = model.LabelValue("node_cpu_usage_seconds_total")
+	nodeMemUsageMetricName      = model.LabelValue("node_memory_working_set_bytes")
+	containerCpuUsageMetricName = model.LabelValue("container_cpu_usage_seconds_total")
+	containerMemUsageMetricName = model.LabelValue("container_memory_working_set_bytes")
+	//Todo: add node/container start time
+
+)
+
+func decodeBatch(samples []*model.Sample, nodeName string) *storage.MetricsBatch {
+	res := &storage.MetricsBatch{
+		Nodes: make([]storage.NodeMetricsPoint, 0),
+		Pods:  make([]storage.PodMetricsPoint, 0),
+	}
+	copiedNodeMetrics := storage.NodeMetricsPoint{}
+	copiedPodMetrics := make(map[apitypes.NamespacedName]map[string]*storage.ContainerMetricsPoint)
+	for _, sample := range samples {
+		//get metrics from sample
+		switch sample.Metric[model.MetricNameLabel] {
+		case nodeCpuUsageMetricName:
+			copiedNodeMetrics.CumulativeCpuUsed = uint64(sample.Value * 1000000000)
+			if sample.Timestamp != 0 {
+				copiedNodeMetrics.Timestamp = time.Unix(0, int64(sample.Timestamp*1000000))
```

Comment:
Why multiple timestamp?

Reply:
Because the data we fetched from /metrics/resource, the unit is millisecond(https://prometheus.io/docs/concepts/data_model/#samples),  but we need  nanosecond.